version: '3'

services:
  django:
    build:
      dockerfile: ./docker//Dockerfile
      context: .
    container_name: django
#    command: python manage.py runserver 0.0.0.0:5000
    entrypoint: /app/docker/runserver.sh
    ports:
      - "5000:5000"
    volumes:
      - media_volume:/app/media
    depends_on:
      pg:
        condition: service_healthy

  celery:
    build:
      context: .
      dockerfile: ./docker//Dockerfile
    command: "celery -A mus --workdir=. worker -l INFO"
    volumes:
      - media_volume:/app/media
    depends_on:
      - django
      - redis
      - pg

  redis:
    restart: unless-stopped
    image: redis:7.0.5-alpine
    expose:
      - 6379

  pg:
    image: postgres
    environment:
      POSTGRES_DB: mus
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
      PGDATA: /var/lib/postgresql/data

    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d mus -U postgres" ]
      interval: 5s
      timeout: 3s
      retries: 5

volumes:
  media_volume: